issue-114 [task-2] Automate the proto file generation by adding the script to the Docker/Make file

a-ajith opened the pull request:
The PR contains:

1. Corrections for the RPC files generation script.
2. New script file to remove the auto-generated RPC files (files generated while development can be removed easily)
3. Docker file changes for generating RPC files for the deployment
4. Makefile changes for enabling RPC file generation and removal.
- **make proto** will generate RPC files for development and testing purposes
- **make protoclean** will remove generated RPC files for clearing the repo
- _make deepclean_ will also remove the RPC files if any generated

issue: #114

amar-shalgar committed:
Adding the helm chart for deploying URplugin

amar-shalgar committed:
Changed the default plugin port to 45007 and 45008

amar-shalgar committed:
Corrected the plugin ur directory name

amar-shalgar committed:
Changed the permissions of files

amar-shalgar committed:
Changed the permissions of files

amar-shalgar committed:
Added the config file changes for URP

amar-shalgar committed:
Removing the build files as these are not required

amar-shalgar committed:
Corrected properties and added new config properties to the yaml file

amar-shalgar committed:
Removed kafka entries from URP and changed the URP image name to urplugin

amar-shalgar committed:
Changed Redisaddress for HA deployment

amar-shalgar committed:
Changed the configurations in URP to include in single dir instead of earlier multiple dir

amar-shalgar committed:
Changed the configurations to read namespace from odimra instead of plugin

amar-shalgar committed:
Removed events properties from URP

amar-shalgar committed:
Added URP plugin docker file

amar-shalgar committed:
Changed the permissions for urplugin script

amar-shalgar committed:
Updated the dockerfile for urp

amar-shalgar committed:
Modified start urplugin script added changes to monitor the service and restart when it has exited

amar-shalgar committed:
Corrected the log path

bharath-b23 committed:
issue-381 : Remove additional setup script of kafka,zookeeper

bharath-b23 committed:
issue-381 : Updated kafka replication configurations

a-ajith committed:
changed the Link to address

a-ajith committed:
added omitempty to the chassis model

a-ajith committed:
Merge branch 'issue-390' of https://github.com/a-ajith/ODIM into issue-390

a-ajith committed:
Merge branch 'development' into issue-390

a-ajith committed:
Merge pull request #423 from a-ajith/issue-390

Issue 390 task 5:  Convert the panic calls in case of handler not implemented to return not implemented error

bharath-b23 committed:
issue-381 : Use derived service name of consul in docker entrypoint scripts

bharath-b23 committed:
issue-381 : Use derived service name of consul in docker entrypoint scripts

Bharath-KKB committed:
Merge pull request #15 from bharath-b-hpe/issue-381-failover-delay

issue-381 : Use derived service name of consul in docker entrypoint scripts

Bharath-KKB committed:
Merge pull request #424 from ODIM-Project/issue-390

Issue 390: GET on Switch Chasis is giving 500 internal service error

Bharath-KKB committed:
Merge pull request #425 from bharath-b-hpe/issue-381

issue-381 : Use derived service name of consul in docker entrypoint scripts

Bharath-KKB committed:
Merge branch 'development' into odim328

bharath-b23 committed:
Merge branch 'development' of https://github.com/ODIM-Project/ODIM into issue-381

bharath-b23 committed:
Merge branch 'issue-381' of https://github.com/ODIM-Project/ODIM into issue-381

tamanna-k committed:
adding route for the individual processor in the GRF plugin

Commit:
Implementation of odim controller cli/tool for deploying ODIM solution

Commit:
Implementation of odim controller cli/tool for deploying ODIM solution

Commit:
fixed unittest workflow

Commit:
fixed unittest workflow

Bharath-KKB committed:
Merge pull request #29 from rahulroshan-kachchap/odim-controller

[Issue-381] Implementation of odim controller cli/tool for deploying ODIM solution

Bharath-KKB committed:
Merge pull request #427 from rahulroshan-kachchap/issue-381

Issue 381 - Implement a CLI for deploying ODIM solution

Commit:
adding logs to task, system and redis files

Commit:
addressign comments

shivacharanms committed:
Merge pull request #429 from lakkundipooja/odim-428

adding logs to task.go, system.go and redis.go  files

Commit:
addressing commens

Bharath-KKB committed:
Merge pull request #17 from bharath-b-hpe/issue-381-nginx-path

issue-381 : Change nginx path used by kubespray

bharath-b23 committed:
issue-381 : Change nginx path used by kubespray

bharath-b23 committed:
issue-381 : Change nginx path used by kubespray

Bharath-KKB committed:
Merge pull request #13 from tamanna-k/odim-4229

issue-399: Adding route for the individual processor in the GRF plugin

Bharath-KKB committed:
Merge pull request #430 from bharath-b-hpe/issue-381

issue-381 : Change nginx path used by kubespray

bharath-b23 committed:
Merge branch 'development' of https://github.com/ODIM-Project/ODIM into issue-381

bharath-b23 committed:
Merge branch 'issue-381' of https://github.com/bharath-b-hpe/ODIM into issue-381

bharath-b23 committed:
Merge branch 'issue-381' of https://github.com/bharath-b-hpe/ODIM into issue-381

bharath-b23 committed:
Merge branch 'development' of https://github.com/ODIM-Project/ODIM into issue-381

bharath-b23 committed:
Merge branch 'issue-381' of https://github.com/ODIM-Project/ODIM into issue-381

bharath-b23 committed:
Merge branch 'issue-381' of https://github.com/ODIM-Project/ODIM into issue-381

Bharath-KKB committed:
Merge pull request #431 from ODIM-Project/issue-381

[Issue 381] - Implement a CLI for deploying ODIM solution

jeevan-kamkar committed:
odim 432:Adding port dmtf model

jeevan-kamkar committed:
odim 432:Adding port dmtf model

Bharath-KKB committed:
Merge pull request #433 from ODIM-Project/odim-432

odim 432:Adding Port model to lib-dmtf package

lakkundipooja committed:
code changes for reset redis master pod

a-ajith committed:
applied go fmt on lib-persistence-manager

Commit:
ran go tools

a-ajith committed:
removed unwanted comments

Bharath-KKB committed:
Merge pull request #7 from lakkundipooja/odim-348-fix

Issue 348- Adding server failed after master redis pod reset

Commit:
added failover time and downtime

Commit:
Merge branch 'odim-348-fix' of https://github.com/lakkundipooja/ODIM into odim-348-fix

Bharath-KKB committed:
Merge pull request #436 from lakkundipooja/odim348

Issue 348- Adding server failed after master redis pod reset

bharath-b23 committed:
issue-434 : Updatedd odim-controller usage logs

Bharath-KKB committed:
Merge pull request #19 from bharath-b-hpe/issue-434-update-logs

issue-434 : Updatedd odim-controller usage logs

Bharath-KKB committed:
Merge branch 'development' into issue-399

Bharath-KKB committed:
Merge branch 'development' into issue-434

shivacharanms committed:
Merge pull request #437 from bharath-b-hpe/issue-434

issue-434 : Updatedd odim-controller usage logs

bharath-b23 committed:
Merge branch 'development' of https://github.com/ODIM-Project/ODIM into issue-434

bharath-b23 committed:
Merge branch 'issue-434' of https://github.com/bharath-b-hpe/ODIM into issue-434

a-ajith committed:
Merge branch 'development' into issue-399

bharath-b-hpe committed:
issue-438 : Update log messages during plugin add and upgrade operation (#20)

* issue-438 : Update log messages during plugin add and upgrade operation

* issue-438 : copy only required helmcharts

Bharath-KKB committed:
Merge pull request #439 from bharath-b-hpe/issue-438

issue-438 : Update log messages during plugin add and upgrade operation

tamanna-k committed:
Merge branch 'development' into issue-399

shivacharanms committed:
Merge pull request #435 from tamanna-k/issue-399

Issue 399: Adding route for the individual processor in the GRF plugin

Bharath-KKB committed:
Merge pull request #21 from bharath-b-hpe/issue-438-jks-warnings

issue-438 : Use latest keytool options to generate kafka and zookeeper JKS

bharath-b23 committed:
issue-438 : Use latest keytool options to generate JKS

bharath-b23 committed:
issue-438 : Use latest keytool options to generate JKS

Bharath-KKB committed:
Merge pull request #440 from bharath-b-hpe/issue-438

issue-438 : Use latest keytool options to generate JKS

shivacharanms committed:
updated default redis configurations

shivacharanms committed:
Merge pull request #442 from shivacharanms/development

updated default redis configurations

Bharath-KKB committed:
Merge branch 'development' into odim328

a-ajith committed:
added retry logic for the fabric plugin contact

a-ajith committed:
made changes to receive the status message for update chassis

irappa-pattar committed:
fix event subscription info deleting automatically

Bharath-KKB committed:
Merge pull request #66 from irappa-pattar/sprint-34-odim-4272

ODIM-4272 fix event subscription of server get deleted automatically

Bharath-KKB committed:
Merge pull request #445 from irappa-pattar/odim-444

Issue-444 fix event subscription info deleting automatically

Commit:
removed consul address suffix from all services

Commit:
removed consul address suffix from all services

Commit:
redfish validator: semantic fix on update service

Bharath-KKB committed:
Merge pull request #48 from a-ajith/issue-443-fix

Issue 443 Fabric plugin token used for chassis not getting updated even after expiry

Bharath-KKB committed:
Merge pull request #18 from shivacharanms/sprint-34-odim-4283

redfish validator: semantic fix on update service

Bharath-KKB committed:
Merge pull request #449 from shivacharanms/odim-448

Odim 448: Semantic fixes on update service

Bharath-KKB committed:
Merge pull request #30 from rahulroshan-kachchap/bruce-4275-fix

[Issue-446] - Removing derived Consul service name for registry address.

rahulroshan-kachchap committed:
Merge branch 'development' into bruce-4275

a-ajith committed:
Merge branch 'development' into issue-443

Bharath-KKB committed:
Merge pull request #450 from rahulroshan-kachchap/bruce-4275

[Issue-446] - Removing derived Consul service name for registry address.

Commit:
added client certifiacte generator script

Commit:
added client certifiacte generator script

Bharath-KKB committed:
Merge branch 'development' into issue-443

Bharath-KKB committed:
Merge pull request #31 from rahulroshan-kachchap/issue-453-script

[Issue-453] - Added client certificate generator script

a-ajith committed:
corrected the namespaces

a-ajith committed:
removed aggregatecollection version from the namespace of metadata

Bharath-KKB committed:
Merge pull request #451 from a-ajith/issue-443

Fix for Issue 443 Fabric plugin token used for chassis not getting updated even after expiry

Bharath-KKB committed:
Merge branch 'development' into issue-453

Bharath-KKB committed:
Merge branch 'development' into odim328

a-ajith committed:
corrected the service root version

Bharath-KKB committed:
Merge pull request #454 from rahulroshan-kachchap/issue-453

[Issue-453] - Added client certificate generator script

a-ajith committed:
Merge pull request #49 from a-ajith/issue-452-fix

issue 452: Namespaces included in metadata are not found in the referenced schema URI

a-ajith committed:
Merge branch 'development' into issue-452

Bharath-KKB reviewed (changes requested):
We dont want to merge it now.

Bharath-KKB committed:
Merge pull request #455 from a-ajith/issue-452

Issue 452: Namespaces included in metadata are not found in the referenced schema URI

jeevan-kamkar committed:
odim 456:Updating the manager links

jeevan-kamkar committed:
odim 456:Updating the manager links

jeevan-kamkar committed:
odim 456:Updating the manager links

Commit:
fixed api service and lib-utils for chassis get

Bharath-KKB committed:
Merge pull request #19 from shivacharanms/odim-4291

fixed api service and lib-utils for chassis get

shivacharanms committed:
Merge pull request #458 from shivacharanms/odim-457

Fix on Chassis resources GET

Bharath-KKB committed:
Merge branch 'development' into odim328

jeevan-kamkar committed:
odim 456:Updating the interface enable property in the port

bharath-b-hpe committed:
issue-459 : Patch request to attach system chassis to Rack is failing (#22)

* issue-459 : Patch request to attach system chassis to Rack is failing

* issue-459 : Patch request to attach system chassis to Rack is failing

* issue-459

bharath-b23 committed:
issue-459 : Replacing southbound URI in fabric repsonse

Bharath-KKB committed:
Update get_chassis_collection_handler.go

Bharath-KKB committed:
Update patch_chassis_handler.go

Bharath-KKB committed:
Update client.go

Bharath-KKB committed:
Merge pull request #460 from bharath-b-hpe/issue-459

issue-459 : Patch request to attach system chassis to Rack is failing

Commit:
structs defined for Zone resource

lakkundipooja committed:
removed ActionInfo from response

 On branch sprint34_bruce_4280
 Changes to be committed:
	modified:   rpc/aggregator.go

Commit:
updated zone struct with links

Bharath-KKB committed:
Merge pull request #462 from lakkundipooja/odim-461

Odim: 461: Fix for Get on /redfish/v1/AggregationService/ResetActionInfo & /redfish/v1/AggregationService/SetDefaultBootOrderActionInfo are giving 404 error

bharath-b23 committed:
issue-464: Update nodeport param in grfplugin service manifest file

irappa-pattar committed:
remove trailing slash in system collection uri @odata.id

Bharath-KKB committed:
Merge branch 'development' into odim328

bharath-b23 committed:
issue-465 : Update openjdk-jre-headless package version in kafka and zookeeper dockerfiles

Bharath-KKB committed:
Merge pull request #23 from bharath-b-hpe/issue-464-grfplugin-fix

issue-464: Update nodeport param in grfplugin service manifest file

Bharath-KKB committed:
Merge pull request #24 from bharath-b-hpe/issue-465-openjdk-version

issue-465 : Update openjdk-jre-headless package version in kafka and zookeeper dockerfiles

jeevan-kamkar committed:
odim 456:Updating the addresspool links

Bharath-KKB committed:
Merge pull request #67 from irappa-pattar/sprint-34-odim-4287

ODIM-4287 remove trailing slash in system collection uri @odata.id

Bharath-KKB committed:
Merge pull request #468 from irappa-pattar/odim-463

Issue-463 remove trailing slash in system collection uri @odata.id

lakkundipooja committed:
added handler for SerialInterface

lakkundipooja committed:
removed ActionInfo from struct

Bharath-KKB committed:
Merge branch 'development' into issue-465

Bharath-KKB committed:
Merge branch 'development' into issue-464

Bharath-KKB committed:
Merge pull request #469 from lakkundipooja/bruce-4280

Odim 461: removed ActionInfo from Action struct

Bharath-KKB committed:
Merge branch 'development' into issue-465

Bharath-KKB committed:
Merge branch 'development' into issue-464

irappa-pattar committed:
add storage structure in lib-dmtf

irappa-pattar committed:
add omitempty to struct fields

Bharath-KKB committed:
Merge pull request #467 from bharath-b-hpe/issue-464

issue-464: Update nodeport param in grfplugin service manifest file

Bharath-KKB committed:
Merge branch 'development' into issue-465

Bharath-KKB committed:
Merge pull request #466 from bharath-b-hpe/issue-465

issue-465 : Update openjdk-jre-headless package version in kafka and zookeeper dockerfiles

irappa-pattar committed:
put common objects in commom.go files

lakkundipooja committed:
ran go fmt

lakkundipooja committed:
Added Serial Interface namespaces in metadata

Bharath-KKB committed:
Merge pull request #9 from lakkundipooja/sprint-34-bruce-4286

Issue fix for bruce 4286: /redfish/v1/Managers/{SystemId}/SerialInterface is giving 404 error

lakkundipooja committed:
Merge branch 'development' into odim-4286

Bharath-KKB committed:
Merge pull request #472 from lakkundipooja/odim-4286

Odim 471:Get on /redfish/v1/Managers/{SystemId}/SerialInterface is giving 404 error

Bharath-KKB committed:
Merge pull request #68 from irappa-pattar/sprint-34-bruce-4322

ODIM-4322 add storage structure in lib-dmtf

chinmayic committed:
Merge branch 'development' into odim-345

chinmayic committed:
Merge branch 'development' into odim328

jeevan-kamkar committed:
odim 456:Adding dmtf model for Endpoint schema

jeevan-kamkar committed:
Switch Link model fix

jeevan-kamkar committed:
Switch Link model fix

Bharath-KKB committed:
Merge pull request #20 from shivacharanms/sprint-35-odim-switchChassis

Switch Link model fix

jeevan-kamkar committed:
Merge branch 'development' into odim-456

shivacharanms committed:
Switch Link model fix (#474)

Co-authored-by: jeevan-kamkar <jeevan.kamkar@hpe.com>
Co-authored-by: Bharath Kumar Bhaskar Kizhur <68339187+Bharath-KKB@users.noreply.github.com>

jeevan-kamkar committed:
Baisc Auth fix on get chassis

jeevan-kamkar committed:
Baisc Auth fix on get chassis

jeevan-kamkar committed:
odim 456:Updating the fabric and reduancy data

Bharath-KKB committed:
Merge pull request #21 from shivacharanms/sprint-35-odim-switchChassis

Baisc Auth fix on get chassis

jeevan-kamkar committed:
odim 456:Updating the fabric and reduancy data

jeevan-kamkar committed:
Merge branch 'development' of https://github.com/shivacharanms/ODIM into development

Commit:
Switch method PATCH to method POST

jeevan-kamkar committed:
Port links model fix

Bharath-KKB committed:
Merge pull request #475 from shivacharanms/development

Fix on Basic auth under plugin contact in fabric scenario  and dmtf model fix for switch and port links

shivacharanms committed:
removing pointer reference to Link in port links model

irappa-pattar committed:
update storagecontrollers name

Bharath-KKB committed:
Merge branch 'development' into odim-470

Bharath-KKB committed:
Merge pull request #476 from irappa-pattar/odim-470

Issue-470 Add storage structure in lib-dmtf

jeevan-kamkar committed:
odim 456:Updating the fmt files

jeevan-kamkar committed:
Merge branch 'development' into odim-456

Bharath-KKB committed:
Merge pull request #479 from ODIM-Project/odim-456

Odim 456:Adding the DMTF models  for Zone and Endpoints

irappa-pattar committed:
remove version in odatatype

Bharath-KKB committed:
Merge pull request #69 from irappa-pattar/sprint-35-odim-4329

ODIM-478 Remove version in odatatype of aggregate collection

irappa-pattar committed:
Merge branch 'development' into odim-478

irappa-pattar committed:
update storage object with pointer

shivacharanms committed:
Merge pull request #70 from irappa-pattar/sprint-35-odim-4449

ODIM-470 update storage object with pointer

shivacharanms committed:
Merge pull request #482 from irappa-pattar/odim-470

Issue-470 update storage object with pointer

Bharath-KKB committed:
Merge branch 'development' into odim-478

Bharath-KKB committed:
Merge pull request #480 from irappa-pattar/odim-478

Issues-478 Remove version in odatatype of aggregate collection

irappa-pattar committed:
expose chassis resource uri in odimra and grf

tamanna-k committed:
fix for update service: simple update

tamanna-k committed:
removing logs used in debugging

shivacharanms committed:
Merge pull request #71 from irappa-pattar/sprint-35-odim-4461

ODIM-4461 Expose chassis resource uri in odimra and grf

bharath-b-hpe committed:
issue-486 : Check both IP address and servicename/FQDN in events to identify the fabric plugin (#25)

shivacharanms committed:
Merge pull request #487 from bharath-b-hpe/issue-486

issue-486 : Check both IP address and servicename/FQDN in events to identify the fabric plugin

shivacharanms committed:
Merge pull request #14 from tamanna-k/odim-4396

Odim 4396: Simple update returns 500 internal server error

shivacharanms committed:
Merge branch 'development' into issue-481

shivacharanms committed:
Merge pull request #488 from tamanna-k/issue-481

Issue 481

shivacharanms committed:
response headers and routes added for fabrics (#22)

Co-authored-by: bharath <bharath@odim.hpe.com>

Bharath-KKB committed:
Merge pull request #490 from shivacharanms/odim-489

response headers and routes added for fabrics (#22)

irappa-pattar committed:
add drives and volumes object

irappa-pattar committed:
update ChassisResource map with chassis resource uris

Bharath-KKB committed:
Merge pull request #72 from irappa-pattar/sprint-35-odim-4472

ODIM-4472 update ChassisResource map with chassis resource uris

irappa-pattar committed:
Merge branch 'development' into odim-484

irappa-pattar committed:
update struct object

Bharath-KKB committed:
Merge pull request #492 from irappa-pattar/odim-484

Issue-484 Expose chassis resource uri in odimra and grf

Bharath-KKB committed:
Merge branch 'development' into odim-345

Bharath-KKB committed:
Merge branch 'development' into odim328

irappa-pattar committed:
use camel case for OData values

irappa-pattar committed:
Merge branch 'development' into odim-470

shivacharanms committed:
Merge pull request #340 from chinmayic/odim328

Issue 328 - addressed a comment "remove unnecessary ` `"

shivacharanms committed:
Merge branch 'development' into odim-345

shivacharanms committed:
Merge branch 'development' into odim-470

shivacharanms committed:
Merge pull request #348 from ODIM-Project/odim-345

Issue 345 - addressed readme doc

shivacharanms committed:
Merge branch 'development' into odim-470

Bharath-KKB committed:
Merge pull request #493 from ODIM-Project/odim-470

Issue-470 Add drive and volume object in lib-dmtf

irappa-pattar committed:
fix search and filter issue

irappa-pattar committed:
add comments

shivacharanms committed:
Merge pull request #73 from irappa-pattar/sprint-35-odim-4386

ODIM-4386 handle search and filter case of processor summary model(Intel(R))

lakkundipooja committed:
code changes in Del function

lakkundipooja committed:
code changes in Del function

lakkundipooja committed:
addressing comments

lakkundipooja committed:
resolving conflicts

lakkundipooja committed:
added changes in svc-task

lakkundipooja committed:
addressign comments

lakkundipooja committed:
addressing comments

skhan-anjum committed:
changes in redis.go

lakkundipooja committed:
removed fmt

lakkundipooja committed:
added * to key

lakkundipooja committed:
ran go tools

Bharath-KKB committed:
Merge branch 'development' into task-issue-407

Bharath-KKB committed:
Merge pull request #477 from ODIM-Project/task-issue-407

Issue 407 fix

irappa-pattar committed:
Merge branch 'development' into development

shivacharanms committed:
Merge pull request #495 from irappa-pattar/development

Issue-494 handle search and filter case of processor summary model(Intel(R))

shivacharanms committed:
Merge pull request #10 from lakkundipooja/bruce_4414

Issue fix: Bruce 4414: Get Individual task With X-Auth should give task details

lakkundipooja committed:
Merge branch 'development' into odim-4414

shivacharanms committed:
Merge pull request #498 from lakkundipooja/odim-4414

Odim 4414

a-ajith committed:
changed select to infinite for loop for continuous detection

Bharath-KKB committed:
Merge pull request #50 from a-ajith/issue-499-fix

[Issue-499] Fix for Adding a new connection method in kube_deploy_nodes.yaml after editing and deleting it doesn't get reflected in ODIM

Bharath-KKB committed:
Merge pull request #501 from a-ajith/issue-499

[Issue-499] Fix for Adding a new connection method in kube_deploy_nodes.yaml after editing and deleting it doesn't get reflected in ODIM

jeevan-kamkar committed:
odim 507:Updating the registries routes (#29)

jeevan-kamkar committed:
odim 507:Updating the get of message registry files

Bharath-KKB committed:
Merge pull request #509 from jeevan-kamkar/odim-507

Odim 507:Updating the routes of message registries

lakkundipooja committed:
added code cahnges for deleting Software inventory (#508)

Co-authored-by: Bharath B <66939670+bharath-b-hpe@users.noreply.github.com>
Co-authored-by: Bharath Kumar Bhaskar Kizhur <68339187+Bharath-KKB@users.noreply.github.com>

irappa-pattar committed:
expose telemetry service related resources (#505)

Co-authored-by: Bharath Kumar Bhaskar Kizhur <68339187+Bharath-KKB@users.noreply.github.com>

a-ajith committed:
resolved conflicts

a-ajith commented:
This PR is out of scope as we are moving away from GoMicro
